lib/Basic/ShellUtility.cpp

```diff
@@ -27,27 +27,32 @@ void appendShellEscapedString(llvm::raw_ostream& os, StringRef string) {
     return;
   }
 
+#if defined(_WIN32)
+  std::string escQuote = "\"";
+#else
+  std::string escQuote = "'";
```

Comment:
Yes, good. Double quotes are dangerous on Unix since they allow shell expansion.

Of course, this isn't strictly correct on Windows either because the escaping rules for cmd.exe are immensely complicated (and that's ON TOP OF the CommandLineToArgvW related escaping)... but of course that's for another PR. :/